src/SFA.DAS.LevyTransferMatching.Web/Controllers/OpportunitiesController.cs

```diff
@@ -1,27 +1,72 @@
 ﻿using Microsoft.AspNetCore.Mvc;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Authorization;
-using SFA.DAS.LevyTransferMatching.Web.Attributes;
 using SFA.DAS.LevyTransferMatching.Web.Orchestrators;
+using SFA.DAS.LevyTransferMatching.Web.Models.Opportunities;
+using SFA.DAS.Authorization.Mvc.Attributes;
+using SFA.DAS.LevyTransferMatching.Web.Authentication;
+using SFA.DAS.LevyTransferMatching.Web.Attributes;
 
 namespace SFA.DAS.LevyTransferMatching.Web.Controllers
 {
-    [AllowAnonymous]
-    [Route("opportunities")]
     [HideAccountNavigation(true)]
     public class OpportunitiesController : Controller
     {
-        private readonly IOpportunitiesOrchestrator _searchFundingOrchestrator;
+        private readonly IAuthenticationService _authenticationService;
+        private readonly IOpportunitiesOrchestrator _opportunitiesOrchestrator;
 
-        public OpportunitiesController(IOpportunitiesOrchestrator searchFundingOrchestrator)
+        public OpportunitiesController(IAuthenticationService authenticationService, IOpportunitiesOrchestrator searchFundingOrchestrator)
         {
-            _searchFundingOrchestrator = searchFundingOrchestrator;
+            _authenticationService = authenticationService;
+            _opportunitiesOrchestrator = searchFundingOrchestrator;
         }
 
         public async Task<IActionResult> Index()
         {
-            var viewModel = await _searchFundingOrchestrator.GetIndexViewModel();
+            var viewModel = await _opportunitiesOrchestrator.GetIndexViewModel();
             return View(viewModel);
         }
+
+        [Route("opportunities/{encodedPledgeId}")]
+        public async Task<IActionResult> Detail(string encodedPledgeId)
+        {
+            var viewModel = await _opportunitiesOrchestrator.GetDetailViewModel(encodedPledgeId);
+
+            if (viewModel != null)
+            {
+                return View(viewModel);
+            }
+            else
+            {
+                return NotFound();
+            }
+        }
+
+        [HttpPost]
+        [Route("opportunities/{encodedPledgeId}")]
+        public IActionResult Detail(OpportunitiesPostRequest opportunitiesPostRequest)
```

Comment:
OK - I can change this again - I changed this previously because you asked me to match the type name to be the same as the _endpoint_, rather than the action name. I'll change this back to the action name, then.

Reply:
Type, variable names, associated validator and tests have been renamed back to what they were originally.